source/common/network/address_impl.cc

```diff
@@ -50,11 +51,19 @@ Address::InstanceConstSharedPtr addressFromSockAddr(const sockaddr_storage& ss,
 InstanceConstSharedPtr addressFromFd(int fd) {
   sockaddr_storage ss;
   socklen_t ss_len = sizeof ss;
-  const int rc = ::getsockname(fd, reinterpret_cast<sockaddr*>(&ss), &ss_len);
+  int rc = ::getsockname(fd, reinterpret_cast<sockaddr*>(&ss), &ss_len);
   if (rc != 0) {
-    throw EnvoyException(fmt::format("getsockname failed for '{}': {}", fd, strerror(errno)));
+    throw EnvoyException(
+        fmt::format("getsockname failed for '{}': ({}) {}", fd, errno, strerror(errno)));
   }
-  return addressFromSockAddr(ss, ss_len);
+  int socket_v6only = 0;
+  socklen_t size_int = sizeof(socket_v6only);
+  rc = ::getsockopt(fd, IPPROTO_IPV6, IPV6_V6ONLY, &socket_v6only, &size_int);
+  if (rc != 0 && errno != EOPNOTSUPP) {
```

Comment:
On second thought, there is a RELEASE_ASSERT that setting this sockopt does not fail, so I think you can treat any failure (ie ignore value of errno) as a failure here.